src/nav_msg_glo.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Dmitry Tatarinov <dmitry.tatarinov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+#include <libswiftnav/nav_msg_glo.h>
+#include <libswiftnav/time.h>
+
+/* local containers for parsing GLO ephemeris*/
+static ephemeris_t g_e;
+static bool done_flag = false;
+static u16 nt;
+static u8 n4;
+static u8 hrs;
+static u8 min;
+static u8 sec;
+
+/* Word Ft (accuracy of measurements), refer to GLO ICD, Table 4.4 */
+const float f_t[] = {1.0f,
+                     2.0f,
+                     2.5f,
+                     4.0f,
+                     5.0f,
+                     7.0f,
+                     10.0f,
+                     12.0f,
+                     14.0f,
+                     16.0f,
+                     32.0f,
+                     64.0f,
+                     128.0f,
+                     256.0f,
+                     512.0f,
+                     -1.0f};
```

Comment:
Be careful about using a negative value in case code does not check this. Instead if a value Ft of 15 is found the whole ephemeris should be marked invalid as it must be a bit error in the data.